src/smart-components/Conversation/dux/__tests__/reducers.spec.js

```diff
@@ -261,4 +262,158 @@ describe('Messages-Reducers', () => {
+        }
+      });
+    };
+    testMessageType('admin', ['admin', 'user', 'file']);
+    testMessageType('user', ['admin', 'user', 'file']);
+    testMessageType('file', ['admin', 'user', 'file']);
+
+    // CustomType
+    const testCustomType = (paramsCustomTypes = [], messageCustomTypeList = []) => {
+      const appliedParamsState = reducers(mockData, {
+        type: actionTypes.MESSAGE_LIST_PARAMS_CHANGED,
+        payload: { customTypes: paramsCustomTypes },
+      });
+      expect(appliedParamsState.messageListParams.customTypes).toEqual(paramsCustomTypes);
+      messageCustomTypeList.forEach((customType) => {
+        const receivedMessage = generateMockMessage(1010);
+        receivedMessage.customType = customType;
+        const receivedMessageState = reducers(appliedParamsState, {
+          type: actionTypes.ON_MESSAGE_RECEIVED,
+          payload: { message: receivedMessage, channel: { url: mockMessage1.channelUrl } },
+        });
+        if (paramsCustomTypes.some((paramsCustomType) => paramsCustomType === customType)) {
+          expect(
+            receivedMessageState.allMessages[receivedMessageState.allMessages.length - 1].messageId
+          ).toEqual(receivedMessage.messageId);
+        } else {
+          expect(
+            receivedMessageState.allMessages[receivedMessageState.allMessages.length - 1].messageId
+          ).not.toEqual(receivedMessage.messageId);
+        }
+      });
+    };
+    testCustomType(['a', 'b', 'c'], ['a', 'd']);
+
+    // SenderUserIds
+    const testSenderUserIds = (paramsSenderUserIds = [], messageSenderIds = []) => {
+      const appliedParamsState = reducers(mockData, {
+        type: actionTypes.MESSAGE_LIST_PARAMS_CHANGED,
+        payload: { senderUserIds: paramsSenderUserIds },
+      });
+      expect(appliedParamsState.messageListParams.senderUserIds).toEqual(paramsSenderUserIds);
+      messageSenderIds.forEach((messageSenderId) => {
+        const receivedMessage = generateMockMessage(1010);
+        receivedMessage.sender = { userId: messageSenderId };
+        const receivedMessageState = reducers(appliedParamsState, {
+          type: actionTypes.ON_MESSAGE_RECEIVED,
+          payload: { message: receivedMessage, channel: { url: mockMessage1.channelUrl } },
+        });
+        if (paramsSenderUserIds.some((paramsSenderUserId) => paramsSenderUserId === messageSenderId)) {
+          expect(
+            receivedMessageState.allMessages[receivedMessageState.allMessages.length - 1].messageId
+          ).toEqual(receivedMessage.messageId);
+        } else {
+          expect(
+            receivedMessageState.allMessages[receivedMessageState.allMessages.length - 1].messageId
+          ).not.toEqual(receivedMessage.messageId);
+        }
+      });
+    };
+    testSenderUserIds(['mark-1', 'mark-2', 'mark-3'], ['mark-1', 'mark-3', 'mark-4']);
+  });
+
+  it('should filter by MESSAGE_LIST_PARAMS_CHANGED when ON_MESSAGE_UPDATED', () => {
```

Comment:
each of these assertion groups should be different tests

Reply:
Do you mean divide it more specifically?

Reply:
yeah, you can still keep them inside this big `it`
->
```
it('should filter by MESSAGE_LIST_PARAMS_CHANGED when ON_MESSAGE_UPDATED', () => {
  it ('should subcase -1')
  it ('should subcase -2') 
})

```

So that we know exactly which subcase is failing